common/trinary/tests/test_trit_tryte.c

```diff
@@ -9,26 +9,47 @@
 #include <unity/unity.h>
 
 #include "common/trinary/trit_tryte.h"
+#include "common/trinary/trit_tryte_sse42.h"
 
-#define TRYTES_IN "AZN9"
-#define EXP 1, 0, 0, -1, 0, 0, -1, -1, -1, 0, 0, 0
+#define TRYTES_IN "AZN9AZN9AZN9AZN9AZN9"
+#define EXP                                                                                                            \
+  1, 0, 0, -1, 0, 0, -1, -1, -1, 0, 0, 0, 1, 0, 0, -1, 0, 0, -1, -1, -1, 0, 0, 0, 1, 0, 0, -1, 0, 0, -1, -1, -1, 0, 0, \
+      0, 1, 0, 0, -1, 0, 0, -1, -1, -1, 0, 0, 0, 1, 0, 0, -1, 0, 0, -1, -1, -1, 0, 0, 0
 
 void test_trits_to_trytes(void) {
   trit_t trits[] = {EXP};
-  tryte_t trytes[4];
+  tryte_t trytes[20];
   tryte_t exp[] = {TRYTES_IN};
-  trits_to_trytes(trits, trytes, 12);
-  TEST_ASSERT_EQUAL_MEMORY(exp, trytes, 4);
+  trits_to_trytes(trits, trytes, 60);
+  TEST_ASSERT_EQUAL_MEMORY(exp, trytes, 20);
 }
 
 void test_trytes_to_trits(void) {
   tryte_t trytes[] = {TRYTES_IN};
-  trit_t trits[12];
+  trit_t trits[60];
   trit_t exp[] = {EXP};
-  trytes_to_trits(trytes, trits, 4);
-  TEST_ASSERT_EQUAL_MEMORY(exp, trits, 12);
+  trytes_to_trits(trytes, trits, 20);
+  TEST_ASSERT_EQUAL_MEMORY(exp, trits, 60);
 }
 
+#if defined(__SSE4_2__)
+void test_trits_to_trytes_sse42(void) {
```

Comment:
Add random tests to cover various lengths and various input data. For each test data convert trits to trytes and back to trits and compare trits, also convert trytes to trits and back to trytes and compare trytes.

Reply:
Should we consider the case of N(which is not a triple value) trit conversion?
I know that `iota_common` has handles the case but it is not taken care in the SIMD acceleration pull request.

Reply:
And maybe the random tests should be raised by another pull request?

Reply:
I think `bench_trit_tryte.c` has covered the length between 16 to 2673, we need to think about a better way to do monkey testing.